sunpy/map/mapcubed.py

```diff
@@ -0,0 +1,590 @@
+        """
+        new_maps = []
+        for i in range(0, len(self)):
+            new_maps.append(self._get_map(i).submap(range_a, range_b))
+        return MapCubed(new_maps)
+
+    def lightcurve(self, location_a, location_b, range_c=None):
+        """Given a pixel index return a lightcurve."""
+        return LightCurve(DataFrame({"{0},{1}".format(location_a, location_b):self.data[location_a,location_b, :]},
+                                    index=self.date))
+
+    @u.quantity_input(dimensions=u.pixel, offset=u.pixel)
+    def superpixel(self, dimensions, offset=(0, 0)*u.pixel, func=np.sum):
+        """Returns a new map consisting of superpixels formed from the
+        original data.  Useful for increasing signal to noise ratio in images.
+        """
+        new_maps = []
+        print(offset)
+        for i in range(0, len(self)):
+            new_maps.append(self._get_map(i).superpixel(dimensions, offset=offset, func=func))
+        return MapCubed(new_maps)
+
+    @u.quantity_input(dimensions=u.pixel)
+    def resample(self, dimensions, method='linear'):
+        """Returns a new Map that has been resampled up or down"""
+        new_maps = []
+        for i in range(0, len(self)):
+            new_maps.append(self._get_map(i).resample(dimensions, method=method))
+        return MapCubed(new_maps)
+
+    def apply_function(self, function, *function_args, **function_kwargs):
+        """
+        Apply a function that operates on the full 3-d data in the mapcube and
+        return a single 2-d map based on that function.
+        :param function: a function that takes a 3-d numpy array as its first
+        argument.
+        :param function_args: function arguments
+        :param function_kwargs: function keywords
+        :return: `sunpy.map.Map`
+            A map that stores the result of applying the function to the 3-d
+            data of the mapcube.
+        """
+        if "mapcube_index" in function_kwargs:
+            mapcube_index = function_kwargs.pop("mapcube_index")
+        else:
+            mapcube_index = 0
+        return Map(function(self.data, *function_args, **function_kwargs), self._meta[mapcube_index])
+
+    def std(self):
+        """
+        Calculate the standard deviation of the data array.
+        """
+        return Map(np.std(self.data, axis=2), self._meta[self.ref_index])
+
+    def mean(self):
+        """
+        Calculate the mean of the data array.
+        """
+        return Map(np.mean(self.data, axis=2), self._meta[self.ref_index])
+
+    def min(self):
+        """
+        Calculate the minimum value of the data array.
+        """
+        return Map(np.min(self.data, axis=2), self._meta[self.ref_index])
+
+    def max(self):
+        """
+        Calculate the maximum value of the data array.
+        """
+        return Map(np.max(self.data, axis=2), self._meta[self.ref_index])
+
+    def plot(self, axes=None, resample=None, annotate=True, interval=200,
```

Comment:
there is a bug here. Need to copy latest code from original mapcube to make this work.